### PR TITLE
Derived Stats implementation and API endpoints

### LINKED ISSUE #485: Enable Derived metrics APIs in the model
The model can provide mean, average and 95%ile for containers, pods, nodes and cluster. This data needs to be exposed via REST endpoints.

### PR DESCRIPTION
**Pull request by afein**
Part 3/3 of #478 

This PR exposes the derived stats that are accessible by the underlying DayStores/StatStores through a set of API endpoints.

Resolves #485 

\cc @vishh @mvdan @mwielgus @piosz

### CODE REVIEW

File: deploy/kube-config/kubedash/heapster-controller.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (vishh) [inline deploy/kube-config/kubedash/heapster-controller.json]: Why is this required? Why don't we make the default config work for kubedash?

File: deploy/kube-config/kubedash/heapster-service.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (vishh) [inline deploy/kube-config/kubedash/heapster-service.json]: Ditto as above. Why is this needed?

File: deploy/kube-config/kubedash/heapster-controller.json (hunk 1 above)
Comment (afein) [inline deploy/kube-config/kubedash/heapster-controller.json]: You're right, I included these configs accidentally in this PR.

File: heapster.go [hunk 3]
```diff
@@ -112,6 +112,7 @@ func doWork() ([]source_api.Source, sinks.ExternalSinkManager, manager.Manager,
 	// Spawn the Model Housekeeping goroutine even if the model is not enabled.
 	// This will allow the model to be activated/deactivated in runtime.
 	// Set the housekeeping period to 2 * argModelResolution + 25 sec
+	// TODO(afein): select a more well-defined housekeeping interval
```
Comment (vishh) [inline heapster.go]: Restating my comment from your previous PR: If we split update to have a separate container and node level metrics storage section and aggregation section, it might be easier to deal with these durations.

File: deploy/kube-config/kubedash/heapster-controller.json (hunk 1 above)
Comment (vishh) [inline deploy/kube-config/kubedash/heapster-controller.json]: is it possible to post incremental changes as separate commits and then squash them at the end? Github makes it hard to track comments if commits are rebased :(

**Comment by vishh** [comment]
LGTM. Thanks @afein for handling all these features! Much appreciated!

File: heapster.go (hunk 3 above)
Comment (afein) [inline heapster.go]: Agree, #528
